core/files/manager.php

```diff
@@ -206,4 +206,60 @@ private function register_actions() {
 	private function reset_assets_data() {
 		delete_option( Page_Assets_Data_Manager::ASSETS_DATA_KEY );
 	}
+
+	/**
+	 * Generate CSS.
+	 *
+	 * Generates CSS for all posts build with Elementor.
+	 *
+	 * @since 3.25.0
+	 * @access public
+	 */
+	public function generate_css() {
+		$batch_size = apply_filters( 'elementor/core/files/generate_css/batch_size', 100 );
+		$processed_posts = 0;
+
+		while ( true ) {
+			$args = [
+				'post_type' => get_post_types(),
+				'posts_per_page' => $batch_size,
+				'meta_query' => [
+					[
+						'key' => Document_Base::BUILT_WITH_ELEMENTOR_META_KEY,
+						'compare' => 'EXISTS',
+					],
+				],
+				'offset' => $processed_posts,
+				'fields' => 'ids',
+			];
+
+			$query = new \WP_Query( $args );
+
+			if ( empty( $query->posts ) ) {
+				break;
+			}
+
+			foreach ( $query->posts as $post_id ) {
+				$document = Plugin::$instance->documents->get_doc_for_frontend( $post_id );
+
+				if ( $document && $document->is_built_with_elementor() ) {
```

Comment:
It's unnecessary because you already do it in the `meta_query`
```suggestion
				if ( $document ) {
```